### PR TITLE
Query execution in mage, module refactor

### PR DESCRIPTION
**Pull request by Josipmrden**
### Description

Periodic and do modules were refactored to work with the Memgraph API without the need to authenticate the client.

### Pull request type

- [X] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

[Delete if this PR doesn't resolve any issues. Link the issue if it does.
](https://github.com/memgraph/memgraph/issues/1650)
Doesn't resolve but related to this

######################################

#### Module/Algorithm
- [X] Core algorithm/module implementation
- [X] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [X] Tests provided (unit / e2e)
- [X] Code documentation
- [X] README short description

### Documentation checklist
- [X] Add the documentation label tag
- [X] Add the bug / feature label tag
- [X] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [X] Write a release note, including added/changed clauses
    - **Periodic and do modules now work with the Memgraph API without the need for a user to authenticate**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [X] Tag someone from docs team in the comments @kgolubic

### CODE REVIEW

File: cpp/periodic_module/periodic.cpp [hunk 1]
```diff
@@ -203,28 +176,22 @@ std::string ConstructFinalQuery(const std::string &running_query, const std::str
...
-    throw std::runtime_error("Unable to connect to client!");
-  }
-  if (!client->Execute(final_query, query_params.AsConstMap())) {
-    throw std::runtime_error("Error while executing periodic iterate!");
-  }
+  auto execution_result = query_execution.ExecuteQuery(final_query, query_params);
 
-  client->DiscardAll();
+  while (execution_result.PullOne()) {
+  }
```
Comment (antepusic) [inline cpp/periodic_module/periodic.cpp:194]: The role of this empty loop is to pull all results, right?

Comment (antepusic) [inline cpp/periodic_module/periodic.cpp:194]: The role of this empty loop is to pull all results, right?

File: cpp/periodic_module/periodic.cpp [hunk 2]
```diff
@@ -20,6 +18,7 @@ const char *kReturnSuccess = "success";
 const char *kReturnNumBatches = "number_of_executed_batches";
 const char *kReturnNumDeletedNodes = "number_of_deleted_nodes";
 const char *kReturnNumDeletedRelationships = "number_of_deleted_relationships";
+const char *kReturnInternalNumDeleted = "num_deleted";
```
Comment (antepusic) [inline cpp/periodic_module/periodic.cpp:21]: Prefer `constexpr const char*` here and for other constants in this file.

**Review by antepusic — COMMENTED** [review]
Good job! Left two quick comments before I may approve.

File: cpp/periodic_module/periodic.cpp (hunk 1 above)
Comment (Josipmrden) [reply cpp/periodic_module/periodic.cpp:194]: yes

**Review by antepusic — APPROVED** [review]

**Pull request merged**